### PR TITLE
Better keep track of renamed features

### LINKED ISSUE #4801: Renamed features appear as "New" in the options
I just noticed that renamed features are being shown as NEW because the renames are just stored as migrations. We should move these renames to an object somewhere and then use the object in the migrations, in the options, `rgh-feature-descriptions`, etc — permanently.

Screen Shot 2

### PR DESCRIPTION
**Pull request by cheap-glitch**
Fix #4801 

## Test URLs
 * `rgh-feature-descriptions`: https://github.com/refined-github/refined-github/blob/21.5.19/source/features/separate-draft-pr-button.tsx
 * `rgh-linkify-features`: https://github.com/refined-github/refined-github/issues/3871

## Screenshot

screenshot-1635516580

(↑ The last feature was recently renamed.)

### CODE REVIEW

**Review by fregante — COMMENTED** [review]
It looks reasonable so far. Let's focus on the bugfix and then it can be improved later. I'm sure there are many places where these renames are useful.

> Check for migrated feature files on build time (so commits like 9589bcd (#4998) should fail since it does not contain a migration)

If that's the only thing missing, you can just follow this review and I'll try to review+merge before the next release.

**Comment by kidonng** [comment]
Quoting myself from https://github.com/refined-github/refined-github/issues/4897#issuecomment-939405024:

> Future implementations should take care of the following things:
> 
> * Feature seen time
> * Check for migrated feature files on build time (so commits like [`9589bcd` (#4998)](https://github.com/refined-github/refined-github/pull/4998/commits/9589bcd47462b07c720e46a0d43d104eb03aa669) should fail since it does not contain a migration)

**Comment by kidonng** [comment]
Would be nice if we have all the past names in `rgh-feature-descriptions`, even just a `title` attribute is enough.

File: source/features/rgh-feature-descriptions.tsx [hunk 1]
```diff
@@ -3,10 +3,12 @@ import elementReady from 'element-ready';
 
 import features from '.';
 import {wrapAll} from '../helpers/dom-utils';
+import {getNewFeatureName} from '../options-storage';
 
 async function init(): Promise<void | false> {
 	const [, currentFeature] = /features\/([^.]+)/.exec(location.pathname)!;
-	const feature = features.meta.find(feature => feature.id === currentFeature);
+	const currentFeatureName = getNewFeatureName(currentFeature);
```
[the remark this answers is not in the document]
Comment (cheap-glitch) [reply source/features/rgh-feature-descriptions.tsx:10]: > Is this why you're doing this here?

It enables the description when viewing the feature file before it was renamed: https://github.com/refined-github/refined-github/blob/21.5.19/source/features/separate-draft-pr-button.tsx
Mostly useful when blaming I guess.

**Comment by cheap-glitch** [comment]
> Check for migrated feature files on build time (so commits like 9589bcd (#4998) should fail since it does not contain a migration)

This might be tricky because we need to detect a file rename, so that probably means querying the GitHub repo in some way?

I agree we should merge this as-is since it fixes the bug and lays a foundation. I'll open up a follow-up PR for the rest.

**Comment by fregante** [comment]
> This might be tricky because we need to detect a file rename, so that probably means querying the GitHub repo in some way?

Nah, we just need to check the latest name is correct. We'll have to check the FROM manually

File: source/features/rgh-feature-descriptions.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (fregante) [reply source/features/rgh-feature-descriptions.tsx:11]: So, yes, what my JS comment said 😃

**Comment by fregante** [comment]
Great! Thanks for fixing this!

PRs welcome for the suggested additional features (for the mid-november version). Or an issue to collect/discuss the suggestions.

Mentioned:

- [ ] > ensure that the final name exists, perhaps in the feature checker file.
- [ ] > modify the "Conversations" link to look for every name the feature ever had
- [ ] > recover past renames. This may be possible by using [githistory.xyz](https://githistory.xyz/) on this file.
- [ ] > This list could also just be a regular JSON file somewhere so it's easily-readable.

**Pull request merged**